src/ServiceControl/Recoverability/Grouping/Groupers/ExceptionTypeAndStackTraceFailureClassifier.cs

```diff
@@ -28,21 +29,28 @@ public string ClassifyFailure(ClassifiableMessageDetails failure)
                 // We need to remove the message in order to make sure the stack trace parser does not get into catastrophic backtracking mode.
                 exceptionStackTrace = exceptionStackTrace.Replace(exception.Message, string.Empty);
             }
-
-            var firstStackTraceFrame = StackTraceParser.Parse(exceptionStackTrace, (frame, type, method, parameterList, parameters, file, line) => new StackFrame
+            try
             {
-                Type = type,
-                Method = method,
-                Params = parameterList,
-                File = file,
-                Line = line
-            }).FirstOrDefault();
+                var firstStackTraceFrame = StackTraceParser.Parse(exceptionStackTrace, (frame, type, method, parameterList, parameters, file, line) => new StackFrame
+                {
+                    Type = type,
+                    Method = method,
+                    Params = parameterList,
+                    File = file,
+                    Line = line
+                }).FirstOrDefault();
 
-            if (firstStackTraceFrame != null)
+                if (firstStackTraceFrame != null)
+                {
+                    return exception.ExceptionType + ": " + firstStackTraceFrame.ToMethodIdentifier();
+                }
+            }
+            catch (RegexMatchTimeoutException)
             {
-                return exception.ExceptionType + ": " + firstStackTraceFrame.ToMethodIdentifier();
+                return GetNonStandardClassification(exception.ExceptionType);
```

Comment:
Do we need this if allowing it to fall through with a comment would execute the same line?

```suggestion
                // Intentionally empty - if the parsing fails, treat it as if exception.Message is null or empty, not throw an exception
```